code/common/strings.cc

```diff
@@ -49,17 +81,41 @@ namespace common::strings
 
     if (size < 1)
       return std::nullopt;
+    #elif defined(__linux__)
+    // Make a copy of the input string in case iconv alters it
+    // +1 to make room for a null-terminator we'll add on the end
+    size_t buf_in_size = utf8.size() + 1;
+    char in_buf[buf_in_size];
+    std::strncpy(in_buf, utf8.data(), utf8.size());
+    // Ensure it's null-terminated: string_view doesn't guarantee null-termination
+    in_buf[utf8.size()] = '\0';
+
+    // Create pointers to be fed into iconv
+    char* ptr_in = static_cast<char*>(in_buf);
+    size_t buf_out_size = buf_in_size; // UTF8 can be a minimum of 1 byte/symbol, wide is 1 element/symbol by def'n
+    wchar_t buf_out[buf_out_size];
+    char* ptr_out = reinterpret_cast<char*>(buf_out); // This is ok, iconv wants a wchar_t* casted to char*
+
+    // utf8 char* to wchar_t*
+    // WCHAR_T should use whatever the system encoding for wchar_t* is
+    // //IGNORE flag will drop anything that can't be WCHAR_T-encoded
+    auto descriptor = iconv_open("WCHAR_T//IGNORE", "UTF-8");
+    iconv(descriptor, &ptr_in, &buf_in_size, &ptr_out, &buf_out_size);
+    iconv_close(descriptor);
+
+    std::wstring out(buf_out);
+    #endif
```

Comment:
Not needed for non-windows

Reply:
Addressed in a1d45580444fd8a62b1644297758291c587b6198